Description: Use GC_reachable_here instead of keepAlive.
 Does it help with isSheddingVertex example?
Author: Doug Torrance <dtorrance@debian.org>
Last-Update: 2022-04-07

--- a/M2/Macaulay2/e/monideal.cpp
+++ b/M2/Macaulay2/e/monideal.cpp
@@ -3,7 +3,6 @@
 #include "monideal.hpp"
 #include "monoid.hpp"
 #include "text-io.hpp"
-#include "finalize.hpp" // for keepAlive
 
 unsigned int MonomialIdeal::computeHashValue() const
 {
@@ -175,7 +174,7 @@
       i++;
       j++;
     }
-  keepAlive(&mi0);
+  GC_reachable_here(&mi0);
   return true;
 }
 
@@ -621,7 +620,7 @@
           }
     }
   MonomialIdeal *result = new MonomialIdeal(get_ring(), new_elems);
-  keepAlive(&J);
+  GC_reachable_here(&J);
   return result;
 }
 
@@ -652,7 +651,7 @@
         new_elems.insert(b);
       }
   MonomialIdeal *result = new MonomialIdeal(get_ring(), new_elems);
-  keepAlive(&J);
+  GC_reachable_here(&J);
   return result;
 }
 
@@ -670,7 +669,7 @@
       new_elems.insert(b);
     }
   MonomialIdeal *result = new MonomialIdeal(get_ring(), new_elems);
-  keepAlive(&J);
+  GC_reachable_here(&J);
   return result;
 }
 
@@ -688,7 +687,7 @@
           result->insert_minimal(b);
         }
     }
-  keepAlive(&J);
+  GC_reachable_here(&J);
   return result;
 }
 
@@ -720,7 +719,7 @@
       delete result;
       result = next_result;
     }
-  keepAlive(&J);
+  GC_reachable_here(&J);
   return result;
 }
 
@@ -799,7 +798,7 @@
       delete result;
       result = next_result;
     }
-  keepAlive(&J);
+  GC_reachable_here(&J);
   return result;
 }
 
